packages/flutter/lib/src/material/drawer.dart

```diff
@@ -644,35 +644,24 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   AlignmentDirectional get _drawerOuterAlignment {
-    switch (widget.alignment) {
-      case DrawerAlignment.start:
-        return AlignmentDirectional.centerStart;
-      case DrawerAlignment.end:
-        return AlignmentDirectional.centerEnd;
-    }
+    return switch (widget.alignment) {
+      DrawerAlignment.start => AlignmentDirectional.centerStart,
+      DrawerAlignment.end   => AlignmentDirectional.centerEnd,
+    };
   }
 
   AlignmentDirectional get _drawerInnerAlignment {
-    switch (widget.alignment) {
-      case DrawerAlignment.start:
-        return AlignmentDirectional.centerEnd;
-      case DrawerAlignment.end:
-        return AlignmentDirectional.centerStart;
-    }
+    return switch (widget.alignment) {
+      DrawerAlignment.start => AlignmentDirectional.centerEnd,
+      DrawerAlignment.end => AlignmentDirectional.centerStart,
+    };
   }
```

Comment:
Theoretically, if you want things even shorter, I think you could even do:

`AlignmentDirectional get _drawerInnerAlignment => switch(...)`

Right?

Reply:
I like using that format for my own projects, but I've been using `return switch(...)` here to follow the [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods):

> only use `=>` when everything, including the function declaration, fits on a single line.

Reply:
oh! That makes sense. Probably the rule is older than switch :P but still valid. Thanks

Reply:
One rule down in the style guide there is also this one: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use--for-inline-callbacks-that-just-return-literals-or-switch-expressions

So, changing this to what Bernardo suggested would be within the rules of the style guide.

Reply:
I'm glad that rule includes `switch` expressions!

I don't think it applies to class getters and methods, but I'll update `focus_traversal.dart` right now 🙂